### PR TITLE
[vertical-pod-autoscaler] Distroless

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

For admission-controller, recommender and updater the base images have been changed to distroless

## Info

```shell
$ ldd admission-controller 
        not a dynamic executable
```

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: vertical-pod-autoscaler
type: feature
summary: For admission-controller, recommender and updater the base images have been changed to distroless
impact: admission-controller, recommender and updater pods will be restarted
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Comment by vitaliy-sn** [comment]
/e2e/abort static;Static;containerd;1.25 6453726941 6453726941-1-con-1-25 /sys/deckhouse-oss/install:pr6099

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

File: modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
-FROM registry.k8s.io/autoscaling/vpa-admission-controller:0.12.0@sha256:997b6efac7bda2b0fd9fe8b54feaeac5fb77cadbbe01a46806f3b0c84a354429 as artifact
-
-FROM $BASE_ALPINE
-COPY --from=artifact /admission-controller /
+ARG BASE_DISTROLESS
+ARG BASE_GOLANG_19_ALPINE
+FROM $BASE_GOLANG_19_ALPINE as artifact
+WORKDIR /src/
+RUN apk add --no-cache git mercurial && \
+    wget https://codeload.github.com/kubernetes/autoscaler/tar.gz/vertical-pod-autoscaler/v0.14.0 -O - | tar -xz --strip-components=1 -C /src/ && \
```
Comment (RomanenkoDenys) [inline modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile:6]: Why do we need mercurial here ?

Comment (RomanenkoDenys) [reply modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile:6]: Also please use git clone instead of wget

Comment (RomanenkoDenys) [reply modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile:6]: Also please use werf format

File: modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+    cd vertical-pod-autoscaler/pkg/admission-controller && \
+    GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-s -w" -o admission-controller
```
Comment (RomanenkoDenys) [inline modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile:8]: We need to change rights to deckhouse:deckhouse 700

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some questions

File: modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile (hunk 2 above)
Comment (vitaliy-sn) [reply modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile]: changed

File: modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile (hunk 1 above)
Comment (vitaliy-sn) [reply modules/302-vertical-pod-autoscaler/images/admission-controller/Dockerfile]: done

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

File: modules/302-vertical-pod-autoscaler/images/admission-controller/werf.inc.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+artifact: {{ .ModuleName }}/admission-controller-artifact
+from: {{ .Images.BASE_GOLANG_19_ALPINE }}
+shell:
+  install:
+    - apk add --no-cache git
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b vertical-pod-autoscaler-0.14.0 https://github.com/kubernetes/autoscaler.git .
```
Comment (RomanenkoDenys) [inline modules/302-vertical-pod-autoscaler/images/admission-controller/werf.inc.yaml:9]: SOURCE_REPO and GOPROXY

File: modules/302-vertical-pod-autoscaler/images/recommender/werf.inc.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  to: /patches
+  stageDependencies:
+    install:
+      - '**/*'
+shell:
+  install:
+    - apk add --no-cache git
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b vertical-pod-autoscaler-0.14.0 https://github.com/kubernetes/autoscaler.git .
```
Comment (RomanenkoDenys) [inline modules/302-vertical-pod-autoscaler/images/recommender/werf.inc.yaml:15]: Same

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
also i think we should use one artifact for building all binaries

File: modules/302-vertical-pod-autoscaler/images/admission-controller/werf.inc.yaml (hunk 3 above)
Comment (name212) [reply modules/302-vertical-pod-autoscaler/images/admission-controller/werf.inc.yaml:9]: And import repository to our gitlab

File: modules/302-vertical-pod-autoscaler/images/updater/werf.inc.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+artifact: {{ .ModuleName }}/updater-artifact
+from: {{ .Images.BASE_GOLANG_19_ALPINE }}
+shell:
+  install:
+    - apk add --no-cache git
+    - mkdir -p /src
+    - cd /src
+    - git clone --depth 1 -b vertical-pod-autoscaler-0.14.0 https://github.com/kubernetes/autoscaler.git .
```
[the remark this answers is not in the document]
Comment (vitaliy-sn) [reply modules/302-vertical-pod-autoscaler/images/updater/werf.inc.yaml]: done

File: modules/302-vertical-pod-autoscaler/images/recommender/werf.inc.yaml (hunk 4 above)
Comment (vitaliy-sn) [reply modules/302-vertical-pod-autoscaler/images/recommender/werf.inc.yaml]: done

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**